appsec-kit-backend/src/main/java/com/vaadin/appsec/backend/AppSecService.java

```diff
@@ -116,7 +116,7 @@ public void init() {
                     + bomMavenFilePath.toAbsolutePath(), e);
         }
 
-        if (isFlow()) {
+        if (isFlow() && configuration.isIncludeNpmDevDependencies()) {
```

Comment:
This additional check is not needed here because if it is true it skips all npm dependencies scan not only the dev dependencies. I'm working on the fix.

Reply:
Fixed.